src/main/java/net/greatstart/services/converters/ProjectConverterService.java

```diff
@@ -0,0 +1,98 @@
+package net.greatstart.services.converters;
+
+import net.greatstart.dto.DtoProject;
+import net.greatstart.model.Category;
+import net.greatstart.model.Project;
+import net.greatstart.model.ProjectDescription;
+import net.greatstart.services.CategoryService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.IOException;
+import java.time.LocalDate;
+
+@Service
+public class ProjectConverterService {
+    private CategoryService categoryService;
+
+    private final Logger logger = LoggerFactory.getLogger(ProjectConverterService.class);
+
+    @Autowired
+    public ProjectConverterService(CategoryService categoryService) {
+        this.categoryService = categoryService;
+    }
+
+    public DtoProject fromProjectToDto(Project project) {
+        DtoProject dtoProject = new DtoProject();
+        if (project != null) {
+            ProjectDescription desc = project.getDesc();
+            dtoProject.setId(project.getId());
+            dtoProject.setName(desc.getName());
+            dtoProject.setGoal(desc.getGoal());
+            dtoProject.setDescription(desc.getDescription());
+            Category category = project.getCategory();
+            if (category != null) {
+                dtoProject.setCategory(category.getName());
+            }
+            dtoProject.setDateStart(desc.getAddStart());
+            dtoProject.setDateAdded(desc.getAddDate());
+            dtoProject.setCost(desc.getCost());
+            dtoProject.setMinInvestment(desc.getMinInvestment());
+            dtoProject.setImage(desc.getImage());
+        }
+        return dtoProject;
+    }
+
+    public void updateProjectFromDto(Project project, DtoProject dtoProject, MultipartFile image) {
+        saveImage(dtoProject, image);
+        ProjectDescription desc = getProjectDescFromDto(dtoProject);
+        project.setDesc(desc);
+        Category category = categoryService.findCategoryByName(dtoProject.getCategory());
+        project.setCategory(category);
+    }
+
+    public Project newProjectFromDto(DtoProject dtoProject, MultipartFile image) {
+        saveImage(dtoProject, image);
+        Project project = new Project();
+        ProjectDescription desc = getProjectDescFromDto(dtoProject);
+        desc.setAddDate(LocalDate.now());
+        Category category = categoryService.findCategoryByName(dtoProject.getCategory());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "dtoProject" is nullable here [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2259)